src/scheduler/ExecGraph.cpp

```diff
@@ -43,6 +43,33 @@ std::set<std::string> getExecGraphHosts(const ExecGraph& graph)
     return getExecGraphHostsForNode(graph.rootNode);
 }
 
+std::vector<std::string> getMpiRankHostsFromExecGraphNode(
```

Comment:
I reckon this method may be too MPI-specific for `ExecGraph.cpp` but I am not sure where to put it.